app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,270 @@
+                if (displayColumnCount > 1) {
+                    val columnDistance = (abs(distance.toFloat()) / columnWidth).roundToInt()
+
+                    MyApp.LogDebug(LOG_TAG,"column distance: $columnDistance")
+
+                    tempColumnIndex = if (distance > 0) {
+                        activeColumnIndex - columnDistance
+                    } else {
+                        activeColumnIndex + columnDistance
+                    }
+                } else {
+                    if (abs(distance) > columnWidth / 4) {
+                        tempColumnIndex = if (distance > 0) {
+                            activeColumnIndex - 1
+                        } else {
+                            activeColumnIndex + 1
+                        }
+                    }
+                }
+                scrollToColumn(tempColumnIndex, false)
+                true
+            } else false
+    }
+
+    override fun onSizeChanged(width: Int, height: Int, oldWidth: Int, oldHeight: Int) {
+        MyApp.LogDebug(LOG_TAG,"onSizeChanged $oldWidth, $oldHeight, $width, $height getMW:$measuredWidth")
+
+        super.onSizeChanged(width, height, oldWidth, oldHeight)
+
+        displayColumnCount = if (roomsCount == NOT_INITIALIZED) {
+            1
+        } else {
+            calculateDisplayColumnCount(
+                measuredWidth,
+                roomsCount,
+                resources.getInteger(R.integer.max_cols),
+                resources.displayMetrics.density,
+                resources.getInteger(R.integer.min_width_dip)
+            )
+        }
+        val newItemWidth = (measuredWidth.toFloat() / displayColumnCount).roundToInt()
+        MyApp.LogDebug(LOG_TAG, """item width: $newItemWidth ${newItemWidth.toFloat() / resources.displayMetrics.density}dp""")
+        setColumnWidth(newItemWidth)
+    }
+
+    override fun onScrollChanged(scrollX: Int, scrollY: Int, oldScrollX: Int, oldScrollY: Int) {
+        super.onScrollChanged(scrollX, scrollY, oldScrollX, oldScrollY)
+        if (roomNames != null) {
+            roomNames!!.scrollTo(scrollX, 0)
+        }
+    }
+
+    fun setChildScroller(scrollView: HorizontalScrollView) {
+        roomNames = scrollView
+    }
+
+    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
+        super.onLayout(changed, left, top, right, bottom)
+        scrollToColumn(activeColumnIndex, true)
+    }
+
+    private fun setColumnWidth(pixels: Int) {
+        MyApp.LogDebug(LOG_TAG, "setColumnWidth $pixels")
+        columnWidth = pixels
+        if (pixels == 0) {
+            return
+        }
+
+        val container = firstChild
+
+        container.children.forEach {
+            // Set width for each Child to columnWidth
+            it.layoutParams.width = columnWidth
+        }
+
+        container.measure(
+            MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED),
+            MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED)
```

Comment:
Can this be de-duplicated?

Reply:
I wouldn't know how, as one defines the height and the other one the width and It seems like there is no version of the function that takes both Parameters as one. It would be possible to add the value as a constant and pass it twice, although that would probably be bad for readability while not really making meaningful improvements

Reply:
OK. Let's leave it as is.